integrated_channels/sap_success_factors/client.py

```diff
@@ -31,28 +31,91 @@ def get_oauth_access_token(url_base, client_id, client_secret, company_id, user_
 
         Returns:
             tuple: Tuple containing access token string and expiration datetime.
+        Raises:
+            HTTPError: If we received a failure response code from SAP SuccessFactors.
+            RequestException: If an unexpected response format was received that we could not parse.
         """
 
         global_sap_config = SAPSuccessFactorsGlobalConfiguration.current()
         url = url_base + global_sap_config.oauth_api_path
 
         response = requests.post(
             url,
-            data={
+            json={
                 'grant_type': 'client_credentials',
                 'scope': {
                     'userId': user_id,
                     'companyId': company_id,
-                    'userType': 'user',
+                    'userType': 'admin',
                     'resourceType': 'learning_public_api',
                 }
             },
-            auth=(client_id, client_secret)
+            auth=(client_id, client_secret),
+            headers={'content-type': 'application/json'}
         )
 
+        response.raise_for_status()
         data = response.json()
-
         try:
-            return data['access_token'], datetime.datetime.utcfromtimestamp(data['expiresIn'])
-        except KeyError:
+            return data['access_token'], datetime.datetime.utcfromtimestamp(data['expires_in'] + int(time.time()))
+        except Exception:
             raise requests.RequestException(response=response)
+
+    def __init__(self, url_base, oauth_access_token, timeout=5):
+        """
+        Instantiate a new client.
+
+        Raises:
+            ValueError: If a URL or access token are not provided.
+        """
+
+        if not url_base:
+            raise ValueError('An API url must be supplied!')
+
+        if not oauth_access_token:
+            raise ValueError('An API access token must be supplied!')
+
+        self.url_base = url_base
+
+        session = requests.Session()
+        session.timeout = timeout
+        session.headers['Authorization'] = 'Bearer {}'.format(oauth_access_token)
+        self.session = session
+        super(SAPSuccessFactorsAPIClient, self).__init__()
```

Comment:
If you're going to create one client and use that many times over a potentially long period of time, you'll probably want some mechanism for refreshing the oauth token when it expires.

e.g. https://github.com/edx/edx-analytics-pipeline/blob/master/edx/analytics/tasks/util/edx_api_client.py#L94-L97